modules/apps/analytics/analytics-message-sender-impl/src/main/java/com/liferay/analytics/message/sender/internal/util/AnalyticsModelUtil.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.analytics.message.sender.internal.util;
+
+import com.liferay.analytics.settings.configuration.AnalyticsConfiguration;
+import com.liferay.analytics.settings.security.constants.AnalyticsSecurityConstants;
+import com.liferay.expando.kernel.model.ExpandoTable;
+import com.liferay.expando.kernel.model.ExpandoTableConstants;
+import com.liferay.petra.function.UnsafeFunction;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Joao Victor Alves
+ */
+public class AnalyticsModelUtil {
+
+	public static List<String> getUserAttributeNames(
+		AnalyticsConfiguration analyticsConfiguration) {
+
+		if (ArrayUtil.isEmpty(analyticsConfiguration.syncedUserFieldNames())) {
+			return _userAttributeNames;
+		}
+
+		List<String> attributeNames = new ArrayList<>();
+
+		attributeNames.add("expando");
+		attributeNames.add("memberships");
+
+		for (String name : _userAttributeNames) {
+			if (ArrayUtil.contains(
+					analyticsConfiguration.syncedUserFieldNames(), name)) {
+
+				attributeNames.add(name);
+			}
+		}
+
+		return attributeNames;
+	}
+
+	public static boolean isCustomField(
+		UnsafeFunction<Long, ExpandoTable, Exception>
```

Comment:
@MrJohn1911 this is too weird. Just change the user to use expandoTableLocalService.fetchExpandoTable() and pass in the ExpandoTable instance. And inside this `isCustomField()` method you check for null.